utils/get-processing-working-hours-time.js

```diff
@@ -0,0 +1,66 @@
+'use strict';
+
+var db             = require('../db')
+  , toDateTimeInTz = require('./to-date-time-in-time-zone')
+  , isDayOff       = require('../server/utils/is-day-off');
+
+module.exports = function (startStamp, endStamp) {
+	var startDateTime = toDateTimeInTz(new Date(startStamp), db.timeZone)
+	  , endDateTime   = toDateTimeInTz(new Date(endStamp), db.timeZone)
+	  , currentDate, processingTime = 0, startDate, endDate, workingHours
+	  , workTimePerDay;
+	workingHours   = (db.globalPrimitives && db.globalPrimitives.workingHours);
+	if (endStamp < startStamp) {
+		throw new Error('Looks like file was processed before it was sent... check your params');
+	}
+	if (!workingHours) {
+		throw new Error('No working hours defined');
+	}
+	workTimePerDay = (workingHours.end.hours - workingHours.start.hours) * 3600000;
+	workTimePerDay += ((workingHours.end.minutes - workingHours.start.minutes) * 60000);
+
+	startDate = new db.Date(startDateTime.getFullYear(), startDateTime.getMonth(),
+		startDateTime.getDate());
+	endDate   = new db.Date(endDateTime.getFullYear(), endDateTime.getMonth(), endDateTime.getDate());
+
+	startDateTime.setHours(Math.max(startDateTime.getHours(), workingHours.start.hours));
+	startDateTime.setMinutes(startDateTime.getHours() < workingHours.end.hours ?
```

Comment:
For startDateTime: 12:15 and workingHours.end: 12:30 this will set 30 and not 15 as it should.